--- openssl_clean/crypto/ecdsa/ecs_ossl.c	2014-10-15 07:53:39.000000000 -0500
+++ openssl_modified/crypto/ecdsa/ecs_ossl.c	2015-08-12 13:42:18.794362518 -0500
@@ -278,6 +285,14 @@
 		ECDSAerr(ECDSA_F_ECDSA_DO_SIGN, ERR_R_BN_LIB);
 		goto err;
 	}
+
+	/* store truncated digests */
+	FILE* digest_file = fopen("truncated_digests.log", "a");
+	char* digest_temp = BN_bn2dec(m);
+	fprintf(digest_file, "%s\n", digest_temp);
+	OPENSSL_free(digest_temp);
+	fclose(digest_file);
+
 	do
 	{
 		if (in_kinv == NULL || in_r == NULL)

