--- openssl_clean/crypto/ecdsa/ecs_ossl.c	2014-10-15 07:53:39.000000000 -0500
+++ openssl_modified/crypto/ecdsa/ecs_ossl.c	2015-08-11 15:43:44.429782271 -0500
@@ -147,10 +147,15 @@
 		/* We do not want timing information to leak the length of k,
 		 * so we compute G*k using an equivalent scalar of fixed
 		 * bit-length. */
-
 		if (!BN_add(k, k, order)) goto err;
 		if (BN_num_bits(k) <= BN_num_bits(order))
 			if (!BN_add(k, k, order)) goto err;
+
+		char* kk = BN_bn2dec(k);
+		FILE* nonce_file = fopen("nonces.log", "a");
+		fprintf(nonce_file, "%s\n", kk);
+		fclose(nonce_file);
+		OPENSSL_free(kk);
 
 		/* compute r the x-coordinate of generator * k */
 		if (!EC_POINT_mul(group, tmp_point, k, NULL, NULL, ctx))
